mithril-aggregator/src/event_store/database/query.rs

```diff
@@ -0,0 +1,60 @@
+use crate::event_store::{Event, EventMessage};
+use chrono::Utc;
+use mithril_common::StdResult;
+use mithril_persistence::sqlite::{Query, SourceAlias, SqLiteEntity, WhereCondition};
+
+/// Queries to insert [EventMessage] to the sqlite database.
+pub struct InsertEventQuery {
+    condition: WhereCondition,
+}
+
+impl InsertEventQuery {
+    pub fn one(message: EventMessage) -> StdResult<Self> {
+        let condition = WhereCondition::new(
+            "(source, action, content, created_at) values (?*, ?*, ?*, ?*)",
+            vec![
+                sqlite::Value::String(message.source),
+                sqlite::Value::String(message.action),
+                sqlite::Value::String(format!(
+                    r#"{{"headers": {}, "content": {}}}"#,
+                    serde_json::to_string(&message.headers)?,
+                    message.content
+                )),
```

Comment:
Since the scope is small I think we can find a middle ground by using the `serde_json::json!` macro:

```suggestion
                sqlite::Value::String(serde_json::to_string(
                    &serde_json::json!({ "headers": message.headers, "content": message.content }),
                )?),
```

Reply:
> Since the scope is small I think we can find a middle ground by using the `serde_json::json!` macro:

Yes that works as well :+1:

Reply:
It's a bit more complicated because `message.content` should not be serialized here. It's already a serialized value done by `TransmitterService` that build the `EventMessage` in function `send_event_message`.
We can make the EventMessage generic with generic type for  ̀content`.
Another possibility, is just to create a generic function that create the EventMessage and keep a serialization of `content`and `headers`. If we do that, attributes should be private and we must remove the `add_header` function (that is never used)

Reply:
I change the type of `content` attribute in `EventMessage` to use a `serde_json::value`.
Then it's possible to use `json!` in the query